flink-scala-shell/src/test/scala/org/apache/flink/api/scala/ScalaShellITCase.scala

```diff
@@ -43,6 +43,13 @@ class ScalaShellITCase extends TestLogger {
   @Rule
   def temporaryFolder = _temporaryFolder
 
+  @After
+  def resetClassLoder(): Unit = {
+    // The Scala interpreter changes current class loader to ScalaClassLoader in every execution
+    // refer to [[ILoop.process()]]. So, we need reset it to original class loader after every Test.
+    Thread.currentThread().setContextClassLoader(classOf[ScalaShellITCase].getClassLoader)
```

Comment:
This resets classloader after each tests instead of after class. Because `TableFactoryService` relies to the correct thread context class loader now. Does this make sense to you @twalthr  @aljoscha  ?

Reply:
The basic change makes sense, yes. But why doesn't it work if we only reset it after all tests? Is it because there are multiple tests where the scala shell test uses the table environment?

Reply:
Yes, there are multiple Table API tests in it.  Because the thread's classloader will be set to `ScalaClassLoader` in each test, and `TableFactoryService` can't find the correct factory from `ScalaClassLoader`, that's why we have to reset thread's classloader after each test.